### ISSUE TITLE
GOG store uses adtraction hyperlink, which gets blocked by adblockers

### ISSUE THREAD

**Issue by PaganDegree** [Heroic-Games-Launcher/HeroicGamesLauncher#3602]
### Describe the bug

the GOG store redirects to https://af.gog.com?as=1838482841, which redirects to (https://track.adtraction.com/t/t?a=1578845458&as=1838482841&t=2&tk=1&url=http://www.gog.com), which redirects to (https://www.gog.com/?utm_campaign=adtraction&utm_medium=affiliate&utm_source=adtraction)

This mess of adtraction based links and redirects seems unnecessary, especially because it breaks if you have adblockers built into your router.

Can you remove the adtraction link and have the store just go to www.gog.com?

### Add logs

```text
(14:35:44) INFO:    [Legendary]:        Legendary location: /opt/Heroic/resources/app.asar.unpacked/build/bin/linux/legendary
(14:35:44) INFO:    [Gog]:              GOGDL location: /opt/Heroic/resources/app.asar.unpacked/build/bin/linux/gogdl
(14:35:44) INFO:    [Connection]:       Connectivity: check-online
(14:35:44) INFO:    [Connection]:       Pinging external endpoints
(14:35:44) INFO:    [Gog]:              Checking for existing gog manifests
(14:35:44) INFO:    [Backend]:          DRM module staus {
  "oimompecagnajdejgnnjijobebaeigek": {
    "name": "Widevine Content Decryption Module",
    "status": "new",
    "version": "4.10.2710.0"
  }
}
(14:35:44) WARNING: [Backend]:          Protocol already registered.
(14:35:44) INFO:    [Legendary]:        Running command: LEGENDARY_CONFIG_PATH=/home/pagandegree/.config/heroic/legendaryConfig/legendary /opt/Heroic/resources/app.asar.unpacked/build/bin/linux/legendary --version
(14:35:44) INFO:    [Gog]:              Running command: GOGDL_CONFIG_PATH=/home/pagandegree/.config/heroic/gogdlConfig /opt/Heroic/resources/app.asar.unpacked/build/bin/linux/gogdl --auth-config-path /home/pagandegree/.config/heroic/gog_store/auth.json --version
(14:35:44) INFO:    [Nile]:             Running command: NILE_CONFIG_PATH=/home/pagandegree/.config/heroic/nile_config /opt/Heroic/resources/app.asar.unpacked/build/bin/linux/nile --version
(14:35:44) INFO:    [Backend]:          
System Information:
CPU: 16x AMD Ryzen 7 7800X3D 8-Core Processor
Memory: 32.81 GB (used: 5 GB)
GPUs:
  GPU 0:
    Name: Advanced Micro Devices, Inc. [AMD/ATI] Raphael
    IDs: D=164e V=1002 SD=8877 SV=1043
    Driver: amdgpu
  GPU 1:
    Name: Advanced Micro Devices, Inc. [AMD/ATI] Navi 31 [Radeon RX 7900 XT/7900 XTX/7900M]
    IDs: D=744c V=1002 SD=5302 SV=1849
    Driver: amdgpu
OS: Linux Mint 21.3 (Virginia) (linux)

The current system is not a Steam Deck
We are not running inside a Flatpak container

Software Versions:
  Heroic: 2.13.0 Dorry & Broggy
  Legendary: 0.20.34 Direct Intervention
  gogdl: 1.0.0
  Nile: 1.0.1 Jonathan Joestar

(14:35:44) INFO:    [Frontend]:         Refreshing undefined Library
(14:35:44) WARNING: [Backend]:          refresh not implemented on Sideload Library Manager
(14:35:44) INFO:    [Legendary]:        Refreshing library...
(14:35:44) INFO:    [Connection]:       Connectivity: online
(14:35:44) INFO:    [Gog]:              Running command: GOGDL_CONFIG_PATH=/home/pagandegree/.config/heroic/gogdlConfig /opt/Heroic/resources/app.asar.unpacked/build/bin/linux/gogdl --auth-config-path /home/pagandegree/.config/heroic/gog_store/auth.json auth
(14:35:44) INFO:    [Backend]:          User Not Found, removing it from Store
(14:35:44) INFO:    [Gog]:              Checking if login is valid
(14:35:44) INFO:    [Gog]:              Running command: GOGDL_CONFIG_PATH=/home/pagandegree/.config/heroic/gogdlConfig /opt/Heroic/resources/app.asar.unpacked/build/bin/linux/gogdl --auth-config-path /home/pagandegree/.config/heroic/gog_store/auth.json auth
(14:35:44) INFO:    [Backend]:          Frontend Ready
(14:35:44) WARNING: [Backend]:          listUpdateableGames not implemented on Sideload Library Manager
(14:35:44) INFO:    [Gog]:              Running command: GOGDL_CONFIG_PATH=/home/pagandegree/.config/heroic/gogdlConfig /opt/Heroic/resources/app.asar.unpacked/build/bin/linux/gogdl --auth-config-path /home/pagandegree/.config/heroic/gog_store/auth.json auth
(14:35:44) INFO:    [Backend]:          Checking for new Heroic Updates
(14:35:44) INFO:    [Backend]:          Checking for current version changelog
(14:35:44) INFO:    [Backend]:          AreWeAntiCheatYet data downloaded
(14:35:45) INFO:    [Gog]:              Saved username to config file
(14:35:46) INFO:    [Winetricks]:       Downloading Winetricks
(14:35:47) INFO:    [Gog]:              Found 0 game(s) to update
(14:35:49) INFO:    [WineDownloader]:   Updating wine versions info
(14:35:49) INFO:    [WineDownloader]:   Fetching upstream information...
(14:35:49) INFO:    [Backend]:          Starting the Download Queue
(14:35:50) INFO:    [WineDownloader]:   wine versions updated
(14:40:46) INFO:    [Connection]:       Connectivity: offline
(14:40:47) INFO:    [Connection]:       Connectivity: check-online
(14:40:47) INFO:    [Connection]:       Pinging external endpoints
(14:40:47) INFO:    [Connection]:       All ping requests failed:
(14:40:47) INFO:    [Connection]:       AggregateError: All promises were rejected
(14:40:52) INFO:    [Connection]:       Pinging external endpoints
(14:40:52) INFO:    [Connection]:       Connectivity: online
```

### Steps to reproduce

1. Go to Heroic
2. Click on Stores, GOG Store
3. See error

### Expected behavior

Open GOG store without issue

### Screenshots

![Screenshot from 2024-02-27 14-37-04](https://github.com/Heroic-Games-Launcher/HeroicGamesLauncher/assets/34691012/3408786c-0e85-43bb-a6db-ff8cfa712644)
![Screenshot from 2024-02-27 14-36-44](https://github.com/Heroic-Games-Launcher/HeroicGamesLauncher/assets/34691012/46fe4603-637d-4acb-b182-e0fd58220492)

### Heroic Version

Latest Stable

### System Information

Linux Mint 21.3

**Comment by imLinguin** [comment]
Duplicate of #3569 I believe

**Comment by awalon** [comment]
This is different to issue #3569. As this case is just caused by adtraction (redirect) link https://af.gog.com (tracking) instead or regular http://gog.com. Works as expected after initial domain was changed to https://gog.com.
Why there's only a tranking URL for GOG and not for Amazon and Epic? Would be great if removed for GOG as well.

**Comment by gregzme17** [comment]
@awalon , how did you modify that asar file.  Text editor or something else?

**Comment by Lariaa** [comment]
@awalon AF stands for affiliate. We teamed up with GoG and get a small amount of every purchase made within Heroic Games Launcher or a browser that uses this link,to support our infrastructure.
Neither Epic nor Amazon has a similar agreement and therefore we use the direct store links for these platforms.

@gregzme17 you don't need to modify Heroic in any way. If you don't want to support us, either use your browser or block the af domain on your system. Heroic will automatically fall back to a non-affiliate store when it can't reach the affiliate-domain

**Comment by gregzme17** [comment]
> @awalon AF stands for affiliate. We teamed up with GoG and get a small amount of every purchase made within Heroic Games Launcher or a browser that uses this link,to support our infrastructure. Neither Epic nor Amazon has a similar agreement and therefore we use the direct store links for these platforms.
> 
> @gregzme17 you don't need to modify Heroic in any way. If you don't want to support us, either use your browser or block the af domain on your system. Heroic will automatically fall back to a non-affiliate store when it can't reach the affiliate-domain

@awalon its not that I dont want to support you, its that I have adblocking built into my opnsense router and the automatic redirect is not working.

**Comment by awalon** [comment]
Sorry but for me and at least some other users (see initial issue) this redirect is a bug, as the redirect isn't working and as a result GoG Store can't be used at all. "af.gog.com" is blocked by local firewall and cannot be modified here, as all tracking stays blocked.

So this modification is currently the one and only option to use this app with GoG.
Another option, which needs some adjusted within application could be a fallback without AF redirect (after some timeout) or an tracking opt out by a onetime fee?

**Comment by Bluejay47** [comment]
Same problem, ad trackers are banned at my network firewall so the GOG store just doesn't work in Heroic on my network. There needs to be a way to disable this and just use the actual GOG site address.

**Comment by imLinguin** [comment]
I think I tracked down the issue. It's a funny one. 
Basically af.gog.com is a middleware that redirects user through `track.adtraction.com` gatweway, which triggers tracking logic.   
The way we handle a fallback is detect a failed connection to `track.adtraction.com` and get a URL that GOG passed down. The thing is the URL is not quite correct, failed connection going through `https://af.gog.com/?as=xxx` returns a `https://www.gog.com:80` which is simply rejected by Heroic's webview.

<details>
<summary>For less technical people</summary>

The URL `https://www.gog.com:80` is invalid because when using protocol https, it already hints port 443.
Here the port 80 is provided, which corresponds to http protocol. You see why it doesn't make much sense

</details>

**Comment by imLinguin** [comment]
Addressed in #3845
